pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
```

Comment:
I got it. But I have a concern that if we put the "only do once during a locker[name]'s lifecycle" logic inside the `pkg/locker`, the package maybe not so universal.